.github/workflows/ci.yml

```diff
@@ -257,34 +263,93 @@ jobs:
-          done
+          # Extract unique CVE IDs from the JSON output
+          jq -r '.Results[].Vulnerabilities[]?.VulnerabilityID' scan-results.json | sort -u > cve-list.txt
+          
+          if [ -s cve-list.txt ]; then
+            echo "found_vulns=true" >> $GITHUB_ENV
+          else
+            echo "found_vulns=false" >> $GITHUB_ENV
+          fi
         env:
           GITHUB_TOKEN: ${{ secrets.ORGANIZATION_TOKEN }}
           TRIVY_JAVA_DB_REPOSITORY: ${{ secrets.TRIVY_JAVA_DB_REPOSITORY }}
           TRIVY_DB_REPOSITORY: ${{ secrets.TRIVY_DB_REPOSITORY }}
+      - name: Upload scan results as artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: trivy-scan-results
+          path: |
+            scan-results.md
+            scan-results.json
+
+      - name: Check for existing issues for each CVE
+        if: env.found_vulns == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.ORGANIZATION_TOKEN }}
+          TARGET_REPO: ${{ inputs.cve-issue-repo || github.repository }}
+        run: |
+          if [[ $directory == '.' ]]; then
+            image=${{ github.event.repository.name }}
+          else
+            image=$directory
+          fi
+          while read -r CVE; do
+            # Check if an issue already exists for this CVE
+            RESPONSE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+                               -H "Accept: application/vnd.github.v3+json" \
+                               "https://api.github.com/repos/$TARGET_REPO/issues?state=open&labels=Trivy")
 
+            # Handle API errors or empty responses
+            if [ -z "$RESPONSE" ] || [ "$RESPONSE" == "[]" ]; then
+              echo "No existing issues found for $CVE (GitHub API returned empty response)"
+              EXISTING_ISSUE=""
+            else
+              EXISTING_ISSUE=$(echo "$RESPONSE" | jq -r --arg cve "$CVE" '.[] | select(.title | contains($cve)) | .title')
+            fi
+
+            # Extract the severity (High or Critical)
+            SEVERITY=$(jq -r --arg cve "$CVE" '.Results[].Vulnerabilities[]? | select(.VulnerabilityID == $cve) | .Severity' scan-results.json | head -n 1)
+            SEVERITY=${SEVERITY:-"Unknown"}
+
+            if [ -z "$EXISTING_ISSUE" ]; then
+              echo "Creating issue for $CVE with severity $SEVERITY in $TARGET_REPO"
+
+              BODY="### Trivy Security Scan Findings in ${{ github.repository }}\n\n- **CVE**: $CVE\n- **Severity**: $SEVERITY\n\n* [CI Run Summary](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})"
+
+              curl -X POST -H "Authorization: token $GITHUB_TOKEN" \
+                   -H "Accept: application/vnd.github.v3+json" \
+                   "https://api.github.com/repos/$TARGET_REPO/issues" \
+                   -d "{\"title\": \"[Trivy] $CVE - Severity: $SEVERITY\", \"body\": \"$BODY\", \"labels\": [\"Trivy\", \"Security\", \"$SEVERITY\"]}"
```

Comment:
Let's use existing approach to create issues - `github.rest.issues` API.

See `notify_failure:` and `close_issue:` for more details.

Btw, `Trivy` and `Security` labels can be replaced by existing `Technical` and `Vulnerability` - similarly to existing tasks (created manually).

Reply:
I can add a config that allows adding extra labels, like "Technical" and `Vulnerability` if that's what you're after.
At the same time, trying to make this as generic as possible (not hardcoding for a particular organization), i think we need to have some sensible and commonly use defaults too.